src/hooks/useAttestation.tsx

```diff
@@ -0,0 +1,36 @@
+import { ProjectIDType, UserDetailsProps } from "@/types/const";
+import { useCallback, useEffect, useState } from "react";
+import { useAccount } from "wagmi";
+import { fetchAttestation } from "@/lib/eas/fetchAttestation";
+import { useParams } from "next/navigation";
+import useProjects from "@/hooks/useProjects";
+
+const useAttestation = () => {
+  const { address } = useAccount();
+  const { id } = useParams<ProjectIDType>();
+  const { projects } = useProjects();
+
+  const [dashboardData, setDashboardData] = useState<UserDetailsProps | null>(
+    null
+  );
+
+  const fetchData = useCallback(async () => {
+    if (projects.length > 0 && address) {
+      let { dashboardData }: any = await fetchAttestation(
+        projects,
```

Comment:
why are you passing a full array of projects into a function to fetch a single attestation?
- feels like a `bloated` parameter that could be made `simpler` right?

philosophy
- less is more
- simple is best
- small is beautiful